src/datasources/notifications/entities/notification-devices.entity.ts

```diff
@@ -0,0 +1,48 @@
+import { NotificationSubscription } from '@/datasources/notifications/entities/notification-subscription.entity';
+import type { UUID } from 'crypto';
+import {
+  Check,
+  Column,
+  Entity,
+  Unique,
+  OneToMany,
+  PrimaryGeneratedColumn,
+} from 'typeorm';
+
+@Entity('push_notification_devices')
+@Unique('device_uuid', ['device_uuid'])
+@Check('device_type', 'device_type IN ("ANDROID", "IOS", "WEB")')
+export class NotificationDevice {
+  @PrimaryGeneratedColumn()
+  id!: number;
+
+  @Column({
+    type: 'varchar',
+    length: 255,
+  })
+  device_type!: string;
```

Comment:
If we can use the enum in the `Check` decorator, this would then be typed that of the enum.

Reply:
IMO, The best solution is to define this field as an enum in the database. But, that's outside the scope of this PR.

Reply:
That would be preferable. If it's any benefit, we already have the `DeviceType` enum in the project.

Reply:
Do you agree with me that it should be refactored in another PR?

Reply:
Yes, that's fine with me.